uaclient/apt.py

```diff
@@ -57,6 +64,21 @@ def assert_valid_apt_credentials(repo_url, username, password):
             os.unlink('/tmp/uaclient-apt-test')
 
 
+def run_apt_command(cmd, error_msg) -> str:
+    """Run an apt command, retrying upon failure APT_RETRIES times.
+
+    :return: stdout from successful run of the apt command.
+    :raise UserFacingError: on issues running apt-cache policy.
+    """
+    try:
+        policy, _err = util.subp(cmd, capture=True, retry_sleeps=APT_RETRIES)
+    except util.ProcessExecutionError as e:
+        if 'Could not get lock /var/lib/dpkg/lock' in str(e.stderr):
```

Comment:
@OddBloke the return values from subprocess.Process.communicate are bytes. I suppose we should convert them in subp before returning stdout stderr to assist us in searching for substrings of the bytes output?

Reply:
@OddBloke I've added decode calls to subp for stderr and stdout so we are now dealiing with strings instead of bytes which allow is to substring search more easily.